cmd/launcher/desktop/desktop.go

```diff
@@ -39,23 +41,22 @@ func handleSignals() {
 
 // continuously monitor for ppid and exit if parent process terminates
 func exitWhenParentGone() {
-	ticker := time.NewTicker(2 * time.Second)
-	defer ticker.Stop()
-
-	f := func() {
-		if os.Getppid() <= 1 {
-			fmt.Println("parent process is gone, exiting")
-			systray.Quit()
-			os.Exit(1)
-		}
-	}
+	for ; true; <-time.NewTicker(2 * time.Second).C {
+		ppid := os.Getppid()
 
-	f()
+		if ppid <= 1 {
+			break
+		}
 
-	for {
-		select {
-		case <-ticker.C:
-			f()
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+		exists, err := process.PidExistsWithContext(ctx, int32(ppid))
+		cancel()
```

Comment:
@directionless , does this look right to you? Can't use `defer` in the for loop since it doesn't get called until the func returns.

Reply:
Do we care about `cancel()`? From https://pkg.go.dev/context#CancelFunc

> A CancelFunc tells an operation to abandon its work. A CancelFunc does not wait for the work to stop. A CancelFunc may be called by multiple goroutines simultaneously. After the first call, subsequent calls to a CancelFunc do nothing.

I we don't need to trigger it outside the timeout, so: `ctx, _ = context.WithTimeout(...)`